lib/manager/pub/readme.md

```diff
@@ -0,0 +1,277 @@
+
+- [**SDK**](https://www.dartlang.org/tools/pub/dependencies#sdk)
+  ```yaml
+  dependencies:
+    flutter_driver:
+      sdk: flutter
+      version: ^0.0.1
+  ```
+
+- [**Version constraints**](https://www.dartlang.org/tools/pub/dependencies#version-constraints)
+  ```yaml
+  dependencies:
+    meta: '>=1.1.0 <2.0.0'
+    flutter_redux: 0.5.0
+  ```
+
+- [**Caret syntax**](https://www.dartlang.org/tools/pub/dependencies#caret-syntax)
+  ```yaml
+  dependencies:
+    redux: ^3.0.0
+  ```
+
+- [**Git packages**](https://www.dartlang.org/tools/pub/dependencies#git-packages)
+  ```yaml
+  dependencies:
+    kittens:
+      git: git://github.com/munificent/kittens.git
+  ```
+  ```yaml
+  dependencies:
+    kittens:
+      git: git@github.com:munificent/kittens.git
+  ```
+  ```yaml
+  dependencies:
+    kittens:
+      git:
+        url: git://github.com/munificent/kittens.git
+        ref: some-branch
+  ```
+  ```yaml
+  dependencies:
+    kittens:
+      git:
+        url: git://github.com/munificent/cats.git
+        path: path/to/kittens
+  ```
+
+- [**Path packages**](https://www.dartlang.org/tools/pub/dependencies#path-packages)
+  ```yaml
+  dependencies:
+    transmogrify:
+      path: /Users/me/transmogrify
+  ```
+
+- [**Hosted packages**](https://www.dartlang.org/tools/pub/dependencies#hosted-packages)
+  ```yaml
+  dependencies:
+    transmogrify:
+      hosted:
+        name: transmogrify
+        url: http://your-package-server.com
+      version: ^1.4.0
+  ```
+
+---
+
+#### Describe which types of dependencies above are supported and which will be implemented in future:
+
+All
+
+## Versioning
+
+#### What versioning scheme do the package files use?
+
+<https://www.dartlang.org/tools/pub/dependencies>
+
+---
+
+#### Does this versioning scheme support range constraints, e.g. `^1.0.0` or `1.x`?
+
+Yes
+
+---
+
+#### Is this package manager used for applications, libraries, or both? If both, is there a way to tell which is which?
+
+Both. No.
+
+---
+
+#### If ranges are supported, are there any cases when Renovate should pin ranges to exact versions if rangeStrategy=auto?
+
+Dev dependencies can always be pinned.
+
+## Lookup
+
+#### Is a new datasource required? Provide details
+
+Not sure.
+
+---
+
+#### Will users need the capability to specify a custom host/registry to look up? Can it be found within the package files, or within other files inside the repository, or would it require Renovate configuration?
+
+Yes
+
+A registry can be specified in _pubspec.yaml_ with:
+
+```yaml
+dependencies:
+  transmogrify:
+    hosted:
+      name: transmogrify
+      url: http://your-package-server.com
+    version: ^1.4.0
+```
+
+---
+
+#### Do the package files contain any "constraints" on the parent language (e.g. supports only v3.x of Python) or platform (Linux, Windows, etc) that should be used in the lookup procedure?
+
+Yes, SDK version can be defined using:
+
+```yaml
+environment:
+  sdk: '>=2.0.0 <3.0.0'
```

Comment:
Does this then influence which package versions that the user downloads? e.g. if there's a package that specifies `>= 4.0.0` for SDK then it would be incompatible with this environment.

Reply:
Yes. 

In the upgrade to Dart to 2.0 there where a lot of packages that became incompatible and some that still are.

Examples:
https://pub.dartlang.org/packages/angular_router
https://pub.dartlang.org/packages/restart
https://pub.dartlang.org/packages/secret_sharing
https://pub.dartlang.org/packages/pixelate

Reply:
This means we’ll use our “compatibility” field to ensure that we don’t propose versions of packages that won’t work with the specified dart version in the spec file.

Reply:
This also means we should provide a way to upgrade the sdk version too